src/overlays/actors/ovl_En_Door/z_en_door.c

```diff
@@ -44,36 +44,66 @@ const ActorInit En_Door_InitVars = {
     (ActorFunc)EnDoor_Draw,
 };
 
+typedef struct {
+    /* 0x00 */ s16 sceneNum;
+    /* 0x02 */ u8 dListIndex;
+    /* 0x04 */ s16 objectId;
+} EnDoorInfo;
+
+typedef enum {
+    /* 0 */ DOOR_DL_GAMEPLAY_KEEP,
+    /* 1 */ DOOR_DL_HIDAN,
+    /* 2 */ DOOR_DL_MIZU,
+    /* 3 */ DOOR_DL_HAKA,
+    /* 4 */ DOOR_DL_GAMEPLAY_FIELD_KEEP,
+    /* 5 */ DOOR_DL_MAX
+} EnDoorDListIndex;
+
 /**
  * Controls which object and display lists to use in a given scene
  */
 static EnDoorInfo sDoorInfo[] = {
-    { SCENE_HIDAN, 1, OBJECT_HIDAN_OBJECTS },
-    { SCENE_MIZUSIN, 2, OBJECT_MIZU_OBJECTS },
-    { SCENE_HAKADAN, 3, OBJECT_HAKA_DOOR },
-    { SCENE_HAKADANCH, 3, OBJECT_HAKA_DOOR },
+    { SCENE_HIDAN, DOOR_DL_HIDAN, OBJECT_HIDAN_OBJECTS },
+    { SCENE_MIZUSIN, DOOR_DL_MIZU, OBJECT_MIZU_OBJECTS },
+    { SCENE_HAKADAN, DOOR_DL_HAKA, OBJECT_HAKA_DOOR },
+    { SCENE_HAKADANCH, DOOR_DL_HAKA, OBJECT_HAKA_DOOR },
     // KEEP objects should remain last and in this order
-    { -1, 0, OBJECT_GAMEPLAY_KEEP },
-    { -1, 4, OBJECT_GAMEPLAY_FIELD_KEEP },
+    { -1, DOOR_DL_GAMEPLAY_KEEP, OBJECT_GAMEPLAY_KEEP },
+    { -1, DOOR_DL_GAMEPLAY_FIELD_KEEP, OBJECT_GAMEPLAY_FIELD_KEEP },
 };
 
 static InitChainEntry sInitChain[] = {
     ICHAIN_U8(targetMode, 0, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_STOP),
 };
 
-static AnimationHeader* sDoorAnims[] = { &gDoor3Anim, &gDoor1Anim, &gDoor4Anim, &gDoor2Anim };
+static AnimationHeader* sDoorAnims[KNOB_ANIM_MAX] = {
+    &gDoor3Anim, // KNOB_ANIM_ADULT_L
+    &gDoor1Anim, // KNOB_ANIM_CHILD_L
+    &gDoor4Anim, // KNOB_ANIM_ADULT_R
+    &gDoor2Anim, // KNOB_ANIM_CHILD_R
+};
 
-static u8 sDoorAnimOpenFrames[] = { 25, 25, 25, 25 };
+static u8 sDoorAnimOpenFrames[KNOB_ANIM_MAX] = {
+    25, // KNOB_ANIM_ADULT_L
+    25, // KNOB_ANIM_CHILD_L
+    25, // KNOB_ANIM_ADULT_R
+    25, // KNOB_ANIM_CHILD_R
+};
 
-static u8 sDoorAnimCloseFrames[] = { 60, 70, 60, 70 };
+static u8 sDoorAnimCloseFrames[KNOB_ANIM_MAX] = {
+    60, // KNOB_ANIM_ADULT_L
+    70, // KNOB_ANIM_CHILD_L
+    60, // KNOB_ANIM_ADULT_R
+    70, // KNOB_ANIM_CHILD_R
+};
 
-static Gfx* sDoorDLists[5][2] = {
-    { gDoorLeftDL, gDoorRightDL },
-    { gFireTempleDoorWithHandleFrontDL, gFireTempleDoorWithHandleBackDL },
-    { gWaterTempleDoorLeftDL, gWaterTempleDoorRightDL },
-    { object_haka_door_DL_0013B8, object_haka_door_DL_001420 },
-    { gFieldDoor1DL, gFieldDoor2DL },
+static Gfx* sDoorDLists[DOOR_DL_MAX][2] = {
+    { gDoorLeftDL, gDoorRightDL },                                         // DOOR_DL_GAMEPLAY_KEEP
+    { gFireTempleDoorWithHandleFrontDL, gFireTempleDoorWithHandleBackDL }, // DOOR_DL_HIDAN
+    { gWaterTempleDoorLeftDL, gWaterTempleDoorRightDL },                   // DOOR_DL_MIZU
+    { object_haka_door_DL_0013B8, object_haka_door_DL_001420 },            // DOOR_DL_HAKA
+    { gFieldDoor1DL, gFieldDoor2DL },                                      // DOOR_DL_GAMEPLAY_FIELD_KEEP
```

Comment:
is `object_haka_door` not named for the same reason, that it's both shadow temple and bottom of the well? Here we have English names though :)

Reply:
Same as the `gDoorXAnim`, I haven't checked the assets

There would also be `gFireTempleDoorWithHandleFrontDL`/Back and `gFieldDoor1DL`/2 to change to Left/Right